### PR TITLE
Track JS errors in GA

### LINKED ISSUE #39: Log js client side errors in our server side logs
We don't track errors in the UI, and rely on users to contact us when things break.

Adding such logging would allow for better discovery of js errors.

### PR DESCRIPTION
**Pull request by tiffon**
Fix #39.

TLDR: Using [raven-js](https://github.com/getsentry/raven-js) to capture errors and context info, then sending it to GA instead of sentry. It's a little out there, but it should be informative.

I did a pretty thorough write-up in [`src/utils/tracking/README.md`](https://github.com/jaegertracing/jaeger-ui/blob/issue-39-track-js-errors/src/utils/tracking/README.md), which lays out the changes in this PR. It might make sense to move some of the README to the Jaeger docs. They need to be updated to reflect `gaTrackingID` ↠ `tracking.gaID` and the addition of `tracking.trackErrors`, anyway.

The changes in the `*.md` files that aren't related to GA tracking can be ignored; they're captured in #188.

The version info generated for GA can also be used to satisfy #113. I'll add a footer in another PR.

I'm currently working on UX event tracking in a different branch.

### CODE REVIEW

**Comment by tiffon** [comment]
@simonrobb Can you look this over when you get the chance?

File: src/utils/tracking/conv-raven-to-ga.js [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable camelcase */
+
+import prefixUrl from '../prefix-url';
+
+const UNKONWN_SYM = { sym: '??', word: '??' };
```
Comment (simonrobb) [inline src/utils/tracking/conv-raven-to-ga.js]: May as well fix this typo (`UNKONWN`) now.

File: src/index.js [hunk 2]
```diff
@@ -12,27 +12,30 @@
...
 import 'u-basscss/css/padding.css';
 import 'u-basscss/css/position.css';
 import 'u-basscss/css/typography.css';
 
-initTracking();
-
 const UI_ROOT_ID = 'jaeger-ui-root';
 
 /* istanbul ignore if */
 if (document && process.env.NODE_ENV !== 'test') {
```
[the remark this answers is not in the document]
Comment (tiffon) [reply src/index.js]: :+1:

File: src/utils/tracking/conv-raven-to-ga.js (hunk 1 above)
Comment (tiffon) [reply src/utils/tracking/conv-raven-to-ga.js]: Done.

File: src/utils/tracking/README.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
+# Google Analytics (GA) Tracking In Jaeger UI
```
Comment (saminzadeh) [inline src/utils/tracking/README.md]: Great write up Joe

Although I'm all for documentation my only comment would be, is it really worth to maintain all this stuff for Jaeger UI? Although this is nice to have, my concern is this information will go stale as its not really part of the core of what Jaeger provides.

File: src/utils/tracking/conv-raven-to-ga.js [hunk 4]
```diff
@@ -0,0 +1,357 @@
+// @flow
```
Comment (saminzadeh) [inline src/utils/tracking/conv-raven-to-ga.js]: I feel like this could be a library in itself outside of Jaeger

File: src/utils/tracking/README.md (hunk 3 above)
Comment (tiffon) [reply src/utils/tracking/README.md]: That's a valid point. But, it really shouldn't need to change much. Seems like low-maintenance docs, to me. 

What do you suggest?

File: scripts/get-tracking-version.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+
+const spawnSync = require('child_process').spawnSync;
```
[the remark this answers is not in the document]
Comment (tiffon) [reply scripts/get-tracking-version.js]: Done.
